test/cpp/cpp_test_util.cpp

```diff
@@ -143,24 +143,30 @@ bool EqualValuesNoElementTypeCheck(at::Tensor tensor1, at::Tensor tensor2) {
 
 void ForEachDevice(absl::Span<const DeviceType> device_types,
                    const std::function<void(const Device&)>& devfn) {
-  const Device* device = GetDefaultDevice();
+  const Device* default_device = GetDefaultDevice();
   if (device_types.empty() ||
-      std::find(device_types.begin(), device_types.end(), device->hw_type) !=
-          device_types.end()) {
-    bridge::SetCurrentDevice(*device);
-    devfn(*device);
+      std::find_if(device_types.begin(), device_types.end(),
```

Comment:
This is changed to `find_if` because it needs to take a predicate.

The new `DeviceType device_type` cannot be compared to `device->hw_type` directly. Now we need to compare `device_type.hw_type` to `default_device->device_type.hw_type` since we are iterating over the new `DeviceTypes`.